From 00165e1a6d5dd61bc153e1352b21ec07fc81245d Mon Sep 17 00:00:00 2001
From: patrick96 <p.ziegler96@gmail.com>
Date: Sun, 11 Feb 2018 21:27:52 +0100
Subject: [PATCH] fix(generators): Add support for eventstruct

Newer xcb-proto commits after the 1.12 release require the 'eventstruct'
key in the output dictionary, otherwise the generator crashes.

I don't see a need for xpp to actually support the eventstruct key and
thus it uses a NOP lambda function
---
 generators/cpp_client.py | 19 ++++++++++---------
 1 file changed, 10 insertions(+), 9 deletions(-)

diff --git a/generators/cpp_client.py b/generators/cpp_client.py
index 20d116f..709e8d8 100644
--- lib/xpp/generators/cpp_client.py
+++ lib/xpp/generators/cpp_client.py
@@ -3130,15 +3130,16 @@ def cpp_type_classes():
 #           }
 
 # Must create an "output" dictionary before any xcbgen imports.
-output = {'open'    : c_open,
-          'close'   : c_close,
-          'simple'  : c_simple, # lambda x, y: None,
-          'enum'    : lambda x, y: None,
-          'struct'  : lambda x, y: None,
-          'union'   : lambda x, y: None,
-          'request' : c_request,
-          'event'   : cpp_event,
-          'error'   : cpp_error,
+output = {'open'          : c_open,
+          'close'         : c_close,
+          'simple'        : c_simple, # lambda x, y: None,
+          'enum'          : lambda x, y: None,
+          'struct'        : lambda x, y: None,
+          'union'         : lambda x, y: None,
+          'request'       : c_request,
+          'event'         : cpp_event,
+          'error'         : cpp_error,
+          'eventstruct'   : lambda x, y: None,
           }
 
 # Boilerplate below this point
